### ISSUE TITLE
New function to adjust validator numbers

### ISSUE THREAD

**Issue by aboka2k** [pactus-project/pactus#494]
Hi, I would like to propose a new function that will allow users to add or remove validators within their Pactus application. This function will greatly assist them in efficiently managing their resources. Thank you.

**Comment by b00f** [comment]
Each validator requires two keys to operate:

1- Validator key: This key is used to sign messages in the consensus algorithm.
2- Reward address: This address is used for receiving rewards.

Due to the use of a Hierarchical Deterministic wallet for address generation, it is not possible to remove addresses once they have been generated. Therefore, if a user wants to remove a validator, they should unbond the validator and later withdraw their stake.

However, users can add additional validators, up to a maximum of 32, for their node. There are two potential solutions:

1-One possible workaround is to add two addresses per new validator and then increase the number of validators inside the configuration file. 

2- Another solution is to re-initialize the node using a seed phrase and, during the initialization phase, add more validators as required. This would allow users to start with the desired number of validators from the beginning.

For both solutions, the address arrangement should change. Currently, we generate addresses in the wallet as follows:

```
1: Validator address 1
2: Validator address 2
3: Validator address 3
4: Reward address 1
5: Reward address 2
6: Reward address 3
```

It should be changed to:

```
1: Validator address 1
2: Reward address 1
3: Validator address 2
4: Reward address 2
5: Validator address 3
6: Reward address 3
```

**Comment by aboka2k** [comment]
1) If its really not possible to remove a validator, maybe can hv an option for them to hide those they dont want?

2) And can we hv a button to automatically add a validator? as it might be too technical or a hassle to most users.

3) I have manually add 4 more validators(8 in total as 4 each for validator/reward). But I have no idea why the first 4 is receiving rewards now. I thought rewards only come from bonding to its own pair or validator?

thank you,

**Comment by b00f** [comment]
> 1. If its really not possible to remove a validator, maybe can hv an option for them to hide those they dont want?

Yes, there is no way to delete and address. maybe we can hide it. This is controversial and need to dig it more.

> 2. And can we hv a button to automatically add a validator? as it might be too technical or a hassle to most users.

User has to restart the nodes. Because the new validator key should be announce to the network and this only happens when node restarts. So in this case modifying a config file is not too hard for normal uses, especially if we have a good documentation. 

> I have manually add 4 more validators(8 in total as 4 each for validator/reward). But I have no idea why the first 4 is receiving rewards now. I thought rewards only come from bonding to its own pair of validator?

This is strange, Please check config file and make sure `num_validators` are correct.

**Comment by aboka2k** [comment]
> > 1. If its really not possible to remove a validator, maybe can hv an option for them to hide those they dont want?
> 
> There is no way to delete an address. Maybe we can hide it, but this is controversial and needs more discussion.
> 
> > 2. And can we hv a button to automatically add a validator? as it might be too technical or a hassle to most users.
> 
> Users have to restart the nodes because the new validator key should be announced to the network, and this only happens when the node restarts. So, in this case, modifying a config file is not too difficult for normal users, especially if we have good documentation.
> 
> > I have manually add 4 more validators(8 in total as 4 each for validator/reward). But I have no idea why the first 4 is receiving rewards now. I thought rewards only come from bonding to its own pair of validator?
> 
> This is strange. Please check the config file and make sure the `num_validators` are correct.

1) i think there is no complication or controversial for this feature. this is jus a very common feature we could find both on pc/mobile where one could hide/unhide some unused data. for us, it might be 'Hide Wallet With 0 Coin'

2) for the 'auto adding' part, maybe we could hv a small utility package together with the program. main program will send values(number of validator etc) to it, and it will close and the utility will create everything automatically and then restart main program and quit itself.

i think this will be the preferable way for everyone, including those tech savvy. as asking them to locate the folder, edit this and then that might deter them from doing it. just take me for eg., im kinda ok with pc, but i still did it wrong :-/

thank you,
